auth.go

```diff
@@ -0,0 +1,53 @@
+package getstream
+
+import (
+	"fmt"
+	"net/http"
+)
+
+type AuthenticationMethod uint32
+
+const (
+	ApplicationAuthentication AuthenticationMethod = iota
+	FeedAuthentication
+)
+
+var authenticators = map[AuthenticationMethod]Authenticator{
+	ApplicationAuthentication: applicationAuthenticator{},
+	FeedAuthentication:        feedAuthenticator{},
+}
+
+var scopeActions = map[string]ScopeAction{
+	"GET":     ScopeActionRead,
+	"OPTIONS": ScopeActionRead,
+	"HEAD":    ScopeActionRead,
+	"POST":    ScopeActionWrite,
+	"PUT":     ScopeActionWrite,
+	"PATCH":   ScopeActionWrite,
+	"DELETE":  ScopeActionDelete,
+}
+
+type Authenticator interface {
+	Authenticate(signer *Signer, request *http.Request, context ScopeContext, feed Feed) error
+}
+
+type applicationAuthenticator struct {
+	key    string
+	secret string
+}
+
+func (a applicationAuthenticator) Authenticate(signer *Signer, request *http.Request, context ScopeContext, feed Feed) error {
+	return signer.SignHTTP(request)
```

Comment:
We want to deprecate http signatures, so this is shouldn't be needed. It's better to use a JWT token with the `"*"` feed permissions.